aten/src/ATen/native/ReduceOps.cpp

```diff
@@ -631,40 +631,32 @@ Tensor &any_out(Tensor &result, const Tensor &self, int64_t dim, bool keepdim) {
   }
 }
 
-Tensor min_values(const Tensor& self, IntArrayRef dims, bool keepdim) {
-  if (dims.size() == 1) {
-    return std::get<0>(self.min(dims[0], keepdim));
-  } else {
-    Tensor result = at::empty({0}, self.options());
-    ScalarType dtype = get_dtype(result, self, {}, true);
-    auto iter = make_reduction("min_values", result, self, dims, keepdim, dtype);
-    TORCH_CHECK(iter.numel() > 0, "min_values on a tensor with no elements is not defined.");
-    min_values_stub(iter.device_type(), iter);
-    return result;
-  }
+Tensor &amin_out(Tensor& result, const Tensor& self, IntArrayRef dim, bool keepdim) {
+  TORCH_CHECK(self.scalar_type() == result.scalar_type(), "Illegal dtype for self, and out:", self.scalar_type(), result.scalar_type());
+  ScalarType dtype = get_dtype(result, self, {}, true);
+  auto iter = make_reduction("amin", result, self, dim, keepdim, dtype);
+  TORCH_CHECK(iter.numel() > 0, "operation does not have an identity");
```

Comment:
There is a test `test_reduction_empty` that requires the message to be precisely `operation does not have an identity`, so this is a common behavior for many reductions.

Reply:
Computation can be independent of result, but it will necessarily have to do same thing that result allocation is already doing, so for this (rare) case we might as well piggyback on it.